CommandLine/emake/EnigmaCallbacks.cpp

```diff
@@ -0,0 +1,92 @@
+#include "EnigmaCallbacks.hpp"
+
+#include <iostream>
+#include <pthread.h>
+#include <unistd.h>
+
+bool CallBack::_isOutputting = false;
+std::ifstream CallBack::_outFile;
+
+void* CallBack::OutputThread(void*)
+{
+    if (_outFile)
```

Comment:
There's tons of UpperCamel in the project here's one example of many https://github.com/enigma-dev/enigma-dev/blob/d9746dfc6d557c4c22c8bd8cf24e02e3272eade0/ENIGMAsystem/SHELL/Universal_System/bufferstruct.h#L94

`Game maker uses lower_func and this project mostly uses  a fair mix of everything. You really need to post style guidelines somewhere because I have no idea what you want me to use?

class OptionsParser or class options_parser or class optionsParser?

void DoThis() or void do_this() or void doThis()?

samething for enums and arguments... I really can't tell what you want